Model/Resolver/EnvironmentVariableResolver.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Semaio\ConfigImportExport\Model\Resolver;
+
+class EnvironmentVariableResolver
+{
+    /**
+     * @var string
+     */
+    private $value;
+
+    /**
+     * Resolve the config value if it's an environment
+     * variable reference.
+     *
+     * @throws \UnexpectedValueException
+     *
+     * @param string $value
+     * @return string|false the original string, or the resolved
+     * string if it's a reference or false on error
+     */
+    public function resolveValue($value)
+    {
+        $this->value = $value;
+        return preg_replace_callback(
+            '/\{env:([^:\}\{]+?)\}/',
+            function ($matches) {
+                $resolvedValue = getenv($matches[1]);
```

Comment:
it may be fine to start with a getenv() call here, but I'd suggest to wrap it with a private method in which you can more easily exchange the resolving of a variable name to a value. you can make it return string|null and give it a second (optional) $default = null parameter.

you can also make the private method do the throw then (or let it pass depending on configuration and need). 

at the end of the day it is normally useful you can then inject an associative array of all environment parameters (with their names as keys) and fill it with $array = getenv() or $_ENV or both etc..

just my 2 cents, in the end you normally want to provide the environment as a whole there (can be easily represented as a PHP associative array).

Reply:
I'm not sure what you're getting at here. Maybe you have a use-case in mind I'm not seeing here..

Reply:
Then leave it as-is. That getting at can escalate quickly otherwise which is normally not good for the first implementation.